### PR TITLE
docs: describe additional database options

### LINKED ISSUE #8168: docs: `databaseOptions` missing in Parse Server Options docs
The `databaseOptions` are not documented in the Parse Server docs. Instead it's just an `Any` type.

The correcting approach is to add a new Type and specify all the `databaseOptions` in that new type. See for example the `Pages` option.

This is a crucial part of the documentation for everyone who is using multiple Parse Server instances, e.g. behind a load balancer. The `databaseOptions.enableSchemaHooks` option is an essential configuration, see https://github.com/parse-community/parse-server/pull/7214.

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

It is not clear that DatabaseOptions are directly passed to the client.

Related issue: #8168
Closes: #8168

### Approach

Adds text explaining docs are directly passed to the client.

### TODOs before merging

- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: src/Options/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ export interface ParseServerOptions {
   /* Options to pass to the database client
   :ENV: PARSE_SERVER_DATABASE_OPTIONS */
   databaseOptions: ?DatabaseOptions;
-  /* Adapter module for the database */
+  /* Adapter module for the database. Additional options are passed directly to the database client */
```
Comment (mtrezza) [inline src/Options/index.js]:
```suggestion
  /* Adapter module for the database; any options that are not explicitly described here are passed directly to the database client. */
```

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.27](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.27)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Pull request merged**